src/arm64/simd.h

```diff
@@ -142,6 +144,40 @@ namespace simdjson::arm64::simd {
     really_inline simd8<L> lookup_16(simd8<L> lookup_table) const {
       return lookup_table.apply_lookup_16_to(*this);
     }
+
+
+    // Copies to 'output" all bytes corresponding to a 0 in the mask (interpreted as a bitset).
+    // Passing a 0 value for mask would be equivalent to writing out every byte to output.
+    // Only the first 16 - count_ones(mask) bytes of the result are significant but 16 bytes
+    // get written.
+    template<typename L>
+    really_inline void compress(uint16_t mask, L * output) const {
```

Comment:
I could imagine having both a `compress(mask)` and `compress(mask, output)` method--the former returning the compressed `simd8<uint8_t>`. (Man, soon enough I'm going to get mad at the number of characters and angle brackets and make a typedef like `simd_u8` / `simd_i8`--I'd really like some analogue of `u8x64` but `u8xmax` looks weird.)

Reply:
By "I could imagine," to be clear, I mean "no need to actually do this, this is excellent and totally fine the way it is, it just tickled my brain."

Reply:
@jkeiser There is a reason it is not coded this way!!! Here is my challenge to you. Go into the haswell version and try to code `compress(mask)`. Don't just stand there: do it. Go! And then, I bet, you'll see the light. :-) I debated whether to document this API choice, and now that it has resurfaced, I will. I will add a comment.

Reply:
Ah, I see it. I keep forgetting that AVX2 is only a partial implementation of 256-bit SIMD.

Reply:
+1